File: dsa/7.linked_list/83_Remove_Duplicates_from_Sorted_List.py
import typing as t
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def deleteDuplicates(self, head: t.Optional[ListNode]) -> t.Optional[ListNode]:
        prev_node: ListNode = None
        pointer_node: ListNode = head
        while pointer_node:
            next_node = pointer_node.next
            if prev_node and prev_node.val == pointer_node.val:
                prev_node.next = next_node
            else:
                prev_node = pointer_node
            pointer_node = next_node

        return head

File: dsa/7.linked_list/83_Remove_Duplicates_from_Sorted_List.java
package linked_list;

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null){
            return null;
        }

        if (head.next == null){
            return head;
        }

        ListNode curNode = head.next;
        ListNode prevNode = head;
        while (curNode != null){
            ListNode surNode = curNode.next;
            if (curNode.val == prevNode.val){
                prevNode.next = surNode;
            }
            else{
                prevNode = curNode;
            }
            curNode = surNode;
        }
        return head;
    }
}
